delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/model/resolvers/ResourceResolutionReport.scala

```diff
@@ -0,0 +1,42 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.resolvers.ResourceResolutionReport.ResolverReport
+
+/**
+  * Report describing how the resource resolution went for each resolver
+  * @param history how the resolution went
+  */
+final case class ResourceResolutionReport(history: List[ResolverReport])
+
+object ResourceResolutionReport {
+
+  /**
+    * Subreport describing how the resolution went for a single resolver
+    */
+  sealed trait ResolverReport extends Product with Serializable {
+
+    /**
+      * @return the resolver
+      */
+    def resolverId: Iri
+
+    /**
+      * @return Causes of the failed attempts to resolve a resource with this resolver
+      */
+    def rejections: List[ResolverResolutionRejection]
+  }
+
+  /**
+    * Report failures for a single resolver
+    */
+  final case class ResolverReportFailed(resolverId: Iri, rejections: List[ResolverResolutionRejection])
```

Comment:
Here you could have Map[ProjectRef, ResolverResolutionRejection] and then keep the rejections untouched